src/test/java/org/broadinstitute/hellbender/tools/CountVariantsIntegrationTest.java

```diff
@@ -15,22 +17,57 @@ public String getTestedClassName() {
         return CountVariants.class.getSimpleName();
     }
 
-    @Test(dataProvider = "filenames")
-    public void testCountVariants(final File fileIn, final long expected) throws Exception {
-        final String[] args = new String[]{
-            "--variant" ,  fileIn.getAbsolutePath(),
-
-        };
-        final Object res = this.runCommandLine(args);
-        Assert.assertEquals(res, expected);
+    @Test(dataProvider = "countVariantsVCFInputs")
+    public void testCountVariants(String fileIn, String moreArgs, long expectedCount) throws Exception {
+        final File ORIG_FILE = new File(getTestDataDir(), fileIn);
+        ArgumentsBuilder ab = new ArgumentsBuilder();
+        ab.add("--variant " + ORIG_FILE.getAbsolutePath());
+        ab.add(moreArgs);
+        final Object res = this.runCommandLine(ab.getArgsArray());
+        Assert.assertEquals(res, expectedCount);
     }
 
-    @DataProvider(name="filenames")
-    public Object[][] filenames() {
+    @DataProvider(name="countVariantsVCFInputs")
+    public Object[][] countVariantsVCFInputs() {
         return new Object[][]{
-                {new File(getTestDataDir(), "count_variants.vcf"), 26L},
-                {new File(getTestDataDir(), "count_variants.blockgz.gz"), 26L},
-                {new File(dbsnp_138_b37_1_65M_vcf), 1375319L},
+                {"count_variants.vcf", "", 26L},
+                {"count_variants.blockgz.gz", "", 26L},
+                {"count_variants_withSequenceDict.vcf", "", 26L},
+                {"count_variants_withSequenceDict.vcf", "-L 1", 14L},
         };
     }
+
+    @Test(expectedExceptions = UserException.class)
+    public void testCountVariants_requiresIndexForInterval() throws Exception {
```

Comment:
I am not a fan of embedding tests for generic engine features (like "variant walkers should require an index when intervals are specified") within the integration tests for random tools like `CountVariants`. I would move these 3 test cases (`testCountVariants_requiresIndexForInterval`, `testCountVariants_missingContigForInterval`, and `testCountVariants_requiresSequenceDictionaryForInterval`) into `FeatureSupportIntegrationTest`